game-core/src/main/java/games/strategy/engine/framework/map/download/DownloadLengthReader.java

```diff
@@ -0,0 +1,103 @@
+package games.strategy.engine.framework.map.download;
+// TODO: move to package games.strategy.engine.framework.map.download
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import games.strategy.engine.framework.system.HttpProxy;
+import lombok.AllArgsConstructor;
+import lombok.extern.java.Log;
+
+/** Can execute an http head request to determine the size of a file download from URI. */
+@Log
+@AllArgsConstructor
+// TODO: testing, break up DownloadUtilsTest to test this component individually from DownloadUtils.
+final class DownloadLengthReader {
+
+  private final Supplier<CloseableHttpClient> httpClientFactory;
+
+  @VisibleForTesting
+  final ConcurrentMap<String, Long> downloadLengthsByUri = new ConcurrentHashMap<>();
```

Comment:
I don't think this map is ever cleared, I really hope this object gets freed at some point.

Reply:
This code is simply moved: https://github.com/triplea-game/triplea/pull/4424/files#diff-557b7a3252d53dbad0eb7925d0ae7dc2L38

Beyond that, I'm not sure if it really matters too much as we do not store very much data in this map, 100 or so elements is a pittance compared to say one image. Regardless, I would prefer to see that kind of memory management be done by managing the scope of the containing object.